core/utils/util.c

```diff
@@ -32,6 +32,7 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
 #include "util.h"
+#include "low_level_platform.h"
```

Comment:
This is part of the solution for reducing code size. `lf_flexpret_support.h` has this in it:

```
// Likewise, fprintf is used to print to `stderr`, but FlexPRET has no `stderr`
// We instead redirect its output to normal printf
#define fprintf(stream, fmt, ...) printf(fmt, ##__VA_ARGS__)
#define vfprintf(fp, fmt, args) vprintf(fmt, args)
```

Since util.c makes a reference to `vfprintf`, including the low level platform (and then `lf_flexpret_support.h`) replaces `vfprintf` with `vprintf` and reduces code size. This solution is quite brittle, though.

Maybe a better solution would be to specify this on the command line, e.g. with some `-Dvfprintf=vprintf`. But then we would have to specify arguments too, since it drops `fp`. And I'm not sure if that can be done on the command line?

Reply:
Maybe just a comment that clarifies this would suffice...